src/app/shared/pipes/unit-filter.pipe.ts

```diff
@@ -0,0 +1,79 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import * as rumble from '../../shared/models/rumble';
+
+export type BuffSearchType = 'both' | 'ability' | 'special';
+
+export interface UnitFilterArgs {
+  filter: string;
+  classes: string[];
+  includeOtherClasses: boolean;
+  types: string[];
+  buffs: string[];
+  buffSearch: BuffSearchType;
+  excludeIds?: number[];
+}
+
+@Pipe({
+  name: 'unitFilter',
+  pure: false,
+})
+export class UnitFilterPipe implements PipeTransform {
```

Comment:
indeed. i even borrowed a bit of the code relative to buff filters.
i feel like pipes make more sense to accomplish this goal (sorting, pagination, and filtering)
as opposed to apply such things in a centralized data source.
for example if we wanted to show two lists of units, each with different filters, i dont think the datasource approach would work (let me know if i misinterpreted the datasource filtering approach)

this pipe can be extended and integrated in the main DB page if you agree

Reply:
ok, still on this topic.. found a bug where if you apply some filters on the DB table and then go to team-builder page via the link, the unit picker will only have units that were filtered in the DB table... this is because i'm using the datasource directly and it still has the filters applied

Reply:
ooh, makes sense. we can reload the data when routing, maybe?